unsigned int tomo::IoLoop::r7_epoll_pass<true, true, true>(int)
--- PRE

+++ POST

@@ -21,157 +21,140 @@

 004a [ 2] test %eax,%eax
 004c [ 4] cmovns %eax,%r13d
 0050 [ 7] mov %r13d,0xc18(%r15)
-0057 [ 6] jle <+0x13f>
-005d [ 7] mov 0x190(%r15),%rax
-0064 [ 3] movslq %r13d,%rdx
-0067 [ 2] xor %ebp,%ebp
-0069 [ 7] lea <rip>(%rip),%r14 # <local-switch-0x130,0x2e0,0x260,0x130,0x130,0x200,0x1d0,0x130,0x120,0x1a0,0x130,0x130,0x130,0x130,0x130,0x130,0x130,0x170>
-0070 [ 7] add %rdx,0x520(%rax)
-0077 [ 7] lea 0xac8(%r15),%rax
-007e [ 5] mov %rax,0x10(%rsp)
-0083 [ 5] lea 0x20(%rsp),%rax
-0088 [ 5] mov %rax,0x8(%rsp)
-008d [ 3] nopl (%rax)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
+0057 [ 2] jle <+0xcb>
+0059 [ 7] mov 0x190(%r15),%rax
+0060 [ 3] movslq %r13d,%rdx
+0063 [ 2] xor %ebp,%ebp
+0065 [ 7] lea <rip>(%rip),%r14 # <local-switch-0xb0,0x2a0,0x100,0xb0,0xb0,0x250,0x230,0xb0,0x1d0,0x210,0xb0,0xb0,0xb0,0xb0,0xb0,0xb0,0xb0,0x1e0>
+006c [ 7] add %rdx,0x520(%rax)
+0073 [ 7] lea 0xac8(%r15),%rax
+007a [ 5] mov %rax,0x10(%rsp)
+007f [ 5] lea 0x20(%rsp),%rax
+0084 [ 5] mov %rax,0x8(%rsp)
+0089 [ 7] nopl 0x0(%rax)
+0090 [ 4] mov 0x4(%rbx),%rsi
+0094 [ 3] mov %rsi,%rax
 0097 [ 4] shr $0x38,%rax
 009b [ 2] cmp $0x11,%al
-009d [ 6] ja <+0x130>
-00a3 [ 4] movslq (%r14,%rax,4),%rax
-00a7 [ 3] add %r14,%rax
-00aa [ 3] notrack jmp *%rax
-00ad [ 2] mov (%rbx),%eax
-00af [ 5] test $0x2019,%eax
-00b4 [ 2] je <+0xba>
-00b6 [ 4] movb $0x0,0x29(%rdx)
-00ba [ 2] test $0x1c,%al
-00bc [ 2] je <+0xe8>
-00be [ 4] cmpb $0x0,0x30(%rdx)
-00c2 [ 5] mov %rdx,0x20(%rsp)
-00c7 [ 2] jne <+0xe8>
-00c9 [ 5] lea 0x20(%rsp),%rsi
-00ce [ 7] lea 0x198(%r15),%rdi
-00d5 [ 4] movb $0x1,0x30(%rdx)
-00d9 [ 5] mov %rdx,0x18(%rsp)
-00de [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
-00e3 [ 5] mov 0x18(%rsp),%rdx
-00e8 [ 3] mov %rdx,%rsi
-00eb [ 3] mov %r15,%rdi
-00ee [ 5] mov %rdx,0x18(%rsp)
-00f3 [ 5] call <tomo::IoLoop::tls_slot_conn(tomo::Client const*)>
-00f8 [ 5] mov 0x18(%rsp),%rdx
-00fd [ 3] test %rax,%rax
-0100 [ 2] je <+0x10c>
-0102 [ 6] movw $0x0,0x3e(%rax)
-0108 [ 4] movb $0x0,0x29(%rdx)
-010c [ 3] mov %rdx,%rsi
-010f [ 3] mov %r15,%rdi
-0112 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
-0117 [ 9] nopw 0x0(%rax,%rax,1)
-0120 [ 3] inc %r12d
-0123 [11] data16 cs nopw 0x0(%rax,%rax,1)
-012e [ 2] xchg %ax,%ax
-0130 [ 2] inc %ebp
-0132 [ 4] add $0xc,%rbx
-0136 [ 3] cmp %r13d,%ebp
-0139 [ 6] jl <+0x90>
-013f [ 5] mov 0x28(%rsp),%rax
-0144 [ 9] sub %fs:0x28,%rax
-014d [ 6] jne <+0x304>
-0153 [ 4] add $0x38,%rsp
-0157 [ 3] mov %r12d,%eax
-015a [ 1] pop %rbx
-015b [ 1] pop %rbp
-015c [ 2] pop %r12
-015e [ 2] pop %r13
-0160 [ 2] pop %r14
-0162 [ 2] pop %r15
-0164 [ 1] ret
-0165 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0170 [ 7] mov 0x190(%r15),%rax
-0177 [ 7] mov 0x188(%r15),%rdi
-017e [ 5] mov 0x10(%rsp),%rdx
-0183 [ 2] inc %ebp
-0185 [ 3] inc %r12d
-0188 [ 4] add $0xc,%rbx
-018c [ 2] mov (%rax),%esi
-018e [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
-0193 [ 3] cmp %r13d,%ebp
-0196 [ 6] jl <+0x90>
-019c [ 2] jmp <+0x13f>
-019e [ 2] xchg %ax,%ax
-01a0 [ 5] mov $0x9,%esi
-01a5 [ 3] mov %r15,%rdi
-01a8 [ 2] inc %ebp
-01aa [ 4] add $0xc,%rbx
-01ae [ 5] call <unsigned int tomo::IoLoop::r7_epoll_accept<true, true>(tomo::UrKind)>
-01b3 [ 3] add %eax,%r12d
-01b6 [ 3] cmp %r13d,%ebp
-01b9 [ 6] jl <+0x90>
-01bf [ 5] jmp <+0x13f>
-01c4 [11] data16 cs nopw 0x0(%rax,%rax,1)
-01cf [ 1] nop
-01d0 [ 5] mov $0x6,%esi
-01d5 [ 3] mov %r15,%rdi
-01d8 [ 2] inc %ebp
-01da [ 4] add $0xc,%rbx
-01de [ 5] call <unsigned int tomo::IoLoop::r7_epoll_accept<true, true>(tomo::UrKind)>
-01e3 [ 3] add %eax,%r12d
-01e6 [ 3] cmp %r13d,%ebp
-01e9 [ 6] jl <+0x90>
-01ef [ 5] jmp <+0x13f>
-01f4 [11] data16 cs nopw 0x0(%rax,%rax,1)
-01ff [ 1] nop
-0200 [ 9] movq $0x0,0x20(%rsp)
-0209 [ 7] nopl 0x0(%rax)
-0210 [ 7] mov 0xbc8(%r15),%edi
-0217 [ 5] mov 0x8(%rsp),%rsi
-021c [ 5] mov $0x8,%edx
-0221 [ 5] call <read@plt>
-0226 [ 4] cmp $0x8,%rax
-022a [ 2] je <+0x210>
-022c [ 7] mov 0x190(%r15),%rax
-0233 [ 2] inc %ebp
-0235 [ 3] inc %r12d
-0238 [ 4] add $0xc,%rbx
-023c [ 7] incq 0x440(%rax)
-0243 [ 3] cmp %r13d,%ebp
-0246 [ 6] jl <+0x90>
-024c [ 5] jmp <+0x13f>
-0251 [11] data16 cs nopw 0x0(%rax,%rax,1)
-025c [ 4] nopl 0x0(%rax)
-0260 [10] movabs $0xffffffffffff,%rax
-026a [ 3] and %rax,%rdx
-026d [ 6] je <+0x130>
-0273 [ 6] mov 0x788(%rdx),%eax
-0279 [ 7] mov 0x190(%r15),%rcx
-0280 [ 2] cmp %eax,(%rcx)
-0282 [ 6] jne <+0x130>
-0288 [ 4] cmpb $0x0,0x33(%rdx)
-028c [ 6] jne <+0x130>
-0292 [ 6] mov 0x784(%rdx),%eax
-0298 [ 3] cmp $0xfffffffe,%eax
-029b [ 6] je <+0x130>
-02a1 [ 7] mov 0x288(%r15),%rax
-02a8 [ 7] mov 0x290(%r15),%rcx
-02af [ 3] cmp %rcx,%rax
-02b2 [ 2] jne <+0x2cd>
-02b4 [ 5] jmp <+0xad>
-02b9 [ 7] nopl 0x0(%rax)
-02c0 [ 4] add $0x28,%rax
-02c4 [ 3] cmp %rax,%rcx
-02c7 [ 6] je <+0xad>
-02cd [ 3] cmp (%rax),%rdx
-02d0 [ 2] jne <+0x2c0>
-02d2 [ 5] jmp <+0x130>
-02d7 [ 9] nopw 0x0(%rax,%rax,1)
-02e0 [ 5] mov $0x1,%esi
-02e5 [ 3] mov %r15,%rdi
-02e8 [ 2] inc %ebp
-02ea [ 4] add $0xc,%rbx
-02ee [ 5] call <unsigned int tomo::IoLoop::r7_epoll_accept<true, true>(tomo::UrKind)>
-02f3 [ 3] add %eax,%r12d
-02f6 [ 3] cmp %r13d,%ebp
-02f9 [ 6] jl <+0x90>
-02ff [ 5] jmp <+0x13f>
-0304 [ 5] call <__stack_chk_fail@plt>
+009d [ 2] ja <+0xb0>
+009f [ 4] movslq (%r14,%rax,4),%rax
+00a3 [ 3] add %r14,%rax
+00a6 [ 3] notrack jmp *%rax
+00a9 [ 7] nopl 0x0(%rax)
+00b0 [ 3] mov %r15,%rdi
+00b3 [ 5] call <tomo::IoLoop::fliplisten_epoll(unsigned long)>
+00b8 [ 3] add %eax,%r12d
+00bb [ 5] nopl 0x0(%rax,%rax,1)
+00c0 [ 2] inc %ebp
+00c2 [ 4] add $0xc,%rbx
+00c6 [ 3] cmp %r13d,%ebp
+00c9 [ 2] jl <+0x90>
+00cb [ 5] mov 0x28(%rsp),%rax
+00d0 [ 9] sub %fs:0x28,%rax
+00d9 [ 6] jne <+0x2b5>
+00df [ 4] add $0x38,%rsp
+00e3 [ 3] mov %r12d,%eax
+00e6 [ 1] pop %rbx
+00e7 [ 1] pop %rbp
+00e8 [ 2] pop %r12
+00ea [ 2] pop %r13
+00ec [ 2] pop %r14
+00ee [ 2] pop %r15
+00f0 [ 1] ret
+00f1 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00fc [ 4] nopl 0x0(%rax)
+0100 [10] movabs $0xffffffffffff,%rdx
+010a [ 3] and %rsi,%rdx
+010d [ 2] je <+0xc0>
+010f [ 6] mov 0x788(%rdx),%eax
+0115 [ 7] mov 0x190(%r15),%rcx
+011c [ 2] cmp %eax,(%rcx)
+011e [ 2] jne <+0xc0>
+0120 [ 4] cmpb $0x0,0x33(%rdx)
+0124 [ 2] jne <+0xc0>
+0126 [ 6] mov 0x784(%rdx),%eax
+012c [ 3] cmp $0xfffffffe,%eax
+012f [ 2] je <+0xc0>
+0131 [ 7] mov 0x288(%r15),%rax
+0138 [ 7] mov 0x290(%r15),%rcx
+013f [ 3] cmp %rcx,%rax
+0142 [ 2] je <+0x162>
+0144 [11] data16 cs nopw 0x0(%rax,%rax,1)
+014f [ 1] nop
+0150 [ 3] cmp (%rax),%rdx
+0153 [ 6] je <+0xc0>
+0159 [ 4] add $0x28,%rax
+015d [ 3] cmp %rax,%rcx
+0160 [ 2] jne <+0x150>
+0162 [ 2] mov (%rbx),%eax
+0164 [ 5] test $0x2019,%eax
+0169 [ 2] je <+0x16f>
+016b [ 4] movb $0x0,0x29(%rdx)
+016f [ 2] test $0x1c,%al
+0171 [ 2] je <+0x19d>
+0173 [ 4] cmpb $0x0,0x30(%rdx)
+0177 [ 5] mov %rdx,0x20(%rsp)
+017c [ 2] jne <+0x19d>
+017e [ 5] lea 0x20(%rsp),%rsi
+0183 [ 7] lea 0x198(%r15),%rdi
+018a [ 4] movb $0x1,0x30(%rdx)
+018e [ 5] mov %rdx,0x18(%rsp)
+0193 [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
+0198 [ 5] mov 0x18(%rsp),%rdx
+019d [ 3] mov %rdx,%rsi
+01a0 [ 3] mov %r15,%rdi
+01a3 [ 5] mov %rdx,0x18(%rsp)
+01a8 [ 5] call <tomo::IoLoop::tls_slot_conn(tomo::Client const*)>
+01ad [ 5] mov 0x18(%rsp),%rdx
+01b2 [ 3] test %rax,%rax
+01b5 [ 2] je <+0x1c1>
+01b7 [ 6] movw $0x0,0x3e(%rax)
+01bd [ 4] movb $0x0,0x29(%rdx)
+01c1 [ 3] mov %rdx,%rsi
+01c4 [ 3] mov %r15,%rdi
+01c7 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
+01cc [ 4] nopl 0x0(%rax)
+01d0 [ 3] inc %r12d
+01d3 [ 5] jmp <+0xc0>
+01d8 [ 8] nopl 0x0(%rax,%rax,1)
+01e0 [ 7] mov 0x190(%r15),%rax
+01e7 [ 7] mov 0x188(%r15),%rdi
+01ee [ 5] mov 0x10(%rsp),%rdx
+01f3 [ 3] inc %r12d
+01f6 [ 2] mov (%rax),%esi
+01f8 [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
+01fd [ 5] jmp <+0xc0>
+0202 [11] data16 cs nopw 0x0(%rax,%rax,1)
+020d [ 3] nopl (%rax)
+0210 [ 5] mov $0x9,%esi
+0215 [ 3] mov %r15,%rdi
+0218 [ 5] call <unsigned int tomo::IoLoop::r7_epoll_accept<true, true>(tomo::UrKind)>
+021d [ 3] add %eax,%r12d
+0220 [ 5] jmp <+0xc0>
+0225 [11] data16 cs nopw 0x0(%rax,%rax,1)
+0230 [ 5] mov $0x6,%esi
+0235 [ 3] mov %r15,%rdi
+0238 [ 5] call <unsigned int tomo::IoLoop::r7_epoll_accept<true, true>(tomo::UrKind)>
+023d [ 3] add %eax,%r12d
+0240 [ 5] jmp <+0xc0>
+0245 [11] data16 cs nopw 0x0(%rax,%rax,1)
+0250 [ 9] movq $0x0,0x20(%rsp)
+0259 [ 7] nopl 0x0(%rax)
+0260 [ 7] mov 0xbc8(%r15),%edi
+0267 [ 5] mov 0x8(%rsp),%rsi
+026c [ 5] mov $0x8,%edx
+0271 [ 5] call <read@plt>
+0276 [ 4] cmp $0x8,%rax
+027a [ 2] je <+0x260>
+027c [ 7] mov 0x190(%r15),%rax
+0283 [ 3] inc %r12d
+0286 [ 7] incq 0x440(%rax)
+028d [ 5] jmp <+0xc0>
+0292 [11] data16 cs nopw 0x0(%rax,%rax,1)
+029d [ 3] nopl (%rax)
+02a0 [ 5] mov $0x1,%esi
+02a5 [ 3] mov %r15,%rdi
+02a8 [ 5] call <unsigned int tomo::IoLoop::r7_epoll_accept<true, true>(tomo::UrKind)>
+02ad [ 3] add %eax,%r12d
+02b0 [ 5] jmp <+0xc0>
+02b5 [ 5] call <__stack_chk_fail@plt>
